mir/src/ir/bus.rs

```diff
@@ -118,6 +118,9 @@ impl Bus {
         self.last = last;
         Ok(())
     }
+    pub fn set_name(&mut self, name: Identifier) {
+        self.name = Some(name);
+    }
```

Comment:
Name could be set only once, right? If so, I'd probably add an assert to catch potential errors if we try to set the name more than once.